Public/Src/Engine/Cache/Artifacts/LocalDiskContentStore.cs

```diff
@@ -926,50 +925,58 @@ private void TryFlushPageCacheToFileSystem(string path)
                 return possibleEnsureFileName.Failure;
             }
 
-            try
-            {
-                // TODO:58494: We shouldn't need to open a new handle here. In fact, that is a race, and why there isn't a RecordContentHashAsync taking a path.
-                // Somebody could change the file since we finished copying from the cache. The cache should instead be enlightened to use the file content table.
-                return FileUtilities.UsingFileHandleAndFileLength(
-                    path.ExpandedPath,
-                    FileDesiredAccess.GenericRead,
-                    FileShare.Read | FileShare.Delete,
-                    FileMode.Open,
-                    FileFlagsAndAttributes.FileFlagOverlapped | FileFlagsAndAttributes.FileFlagOpenReparsePoint,
-                    (handle, length) =>
+
+            // TODO:58494: We shouldn't need to open a new handle here. In fact, that is a race, and why there isn't a RecordContentHashAsync taking a path.
+            // Somebody could change the file since we finished copying from the cache. The cache should instead be enlightened to use the file content table.
+            return Helpers.RetryOnFailure(

```

Comment:
@narasamdya : can you confirm that `TryOpenAndTrackPathAsync` gets called only when we believe that the file exists on disk?

Reply:
Yes, `TryOpenAndTrackPathAsync`  implicitly assume that the file exists. For example, it's only called if file materialization from cache succeeds, which means the file gets materialized on disk.

I'm happy to discuss the issue offline in case I misunderstand something. Let's discuss this first (waiting on this)

---
In reply to: [278650019](https://github.com/Microsoft/BuildXL/pull/202#discussion_r278650019) [](ancestors = 278650019)